src/sentry/search/snuba/executors.py

```diff
@@ -1192,75 +1240,163 @@ def get_suggested(self, search_filter: SearchFilter, joined_entity: Entity) -> C
             conditions = conditions + [suspect_commit_user, ownership_rule_user, codeowner_user]
 
         if team_ids:
             ownership_rule_team = Condition(
-                Column("owner_ownership_rule_team_id", attr_entity), Op.IN, team_ids
+                Column("owner_ownership_rule_team_id", attr_entity), operator, team_ids
             )
             codowner_team = Condition(
-                Column("owner_codeowners_team_id", attr_entity), Op.IN, team_ids
+                Column("owner_codeowners_team_id", attr_entity), operator, team_ids
             )
-            conditions = conditions + [ownership_rule_team, codowner_team]
-
-        if None in search_filter.value.raw_value:
-            # neither assigned to team or user
-            suspect_commit_user = Condition(
-                Column("owner_suspect_commit_user_id", attr_entity), Op.IS_NULL, None
-            )
-            ownership_rule_user = Condition(
-                Column("owner_ownership_rule_user_id", attr_entity), Op.IS_NULL, None
-            )
-            ownership_rule_team = Condition(
-                Column("owner_ownership_rule_team_id", attr_entity), Op.IS_NULL, None
-            )
-            codeowner_user = Condition(
-                Column("owner_codeowners_user_id", attr_entity), Op.IS_NULL, None
-            )
-            codowner_team = Condition(
-                Column("owner_codeowners_team_id", attr_entity), Op.IS_NULL, None
-            )
-            conditions.append(
-                BooleanCondition(
-                    op=BooleanOp.AND,
+            # need to explicitly allow for null values if we are negating
+            if search_filter.is_negation:
+                ownership_rule_team = BooleanCondition(
+                    op=BooleanOp.OR,
                     conditions=[
-                        suspect_commit_user,
-                        ownership_rule_user,
                         ownership_rule_team,
-                        codeowner_user,
+                        Condition(
+                            Column("owner_ownership_rule_team_id", attr_entity),
+                            null_check_operator,
+                            None,
+                        ),
+                    ],
+                )
+                codowner_team = BooleanCondition(
+                    op=BooleanOp.OR,
+                    conditions=[
                         codowner_team,
+                        Condition(
+                            Column("owner_codeowners_team_id", attr_entity),
+                            null_check_operator,
+                            None,
+                        ),
                     ],
                 )
-            )
+            conditions = conditions + [ownership_rule_team, codowner_team]
+
+        # need to check for unsuggested issues if we are checking for none
+        if check_for_none:
+            conditions.append(self.get_unsuggested(search_filter.is_negation))
```

Comment:
worth adding a test for this branch?

Reply:
@JoshFerge good catch! It's actually dead code that's not needed.